### PR TITLE
TRITON-1964 sync with upstream and document syncing with upstream procedure

### PR DESCRIPTION
**Pull request by bahamat**

### CODE REVIEW

File: README [hunk 1]
```diff
@@ -26,3 +26,79 @@ systems.  This utility is used by util/genliso and util/gensdsk.  You
...
+configurations need to be tested.
+
+* Booting from the USB in EFI mode
+* Booting from the USB in legacy mode
+* PXE booting from the hardware NIC and chain loading ipxe over the network
+  in EFI mode
+* PXE booting from the hardware NIC and chain loading ipxe over the network
+  in legacy mode
+
+To install the new ipxe files on an existing Triton USB, downlaod the ipxe
```
Comment (jlevon) [inline README]: nit "download"

File: README [hunk 2]
```diff
@@ -26,3 +26,79 @@ systems.  This utility is used by util/genliso and util/gensdsk.  You
...
+You're done. After testing, get code review, integration approval, then merge.
+
+## Testing changes
+
+**Warning:** If there's a problem with the boot files the compute node(s) may
+not be able to boot at all. It is recomended to perform testing on hardware
+that you can access physically, or can withstand being down for an extended
+period of time in case something goes wrong.
+
+You'll need to test this on as many different hardware platforms and as many
```
Comment (jlevon) [inline README:98]: worth mentioning vmware and (linux) kvm?

File: README [hunk 3]
```diff
@@ -26,3 +26,79 @@ systems.  This utility is used by util/genliso and util/gensdsk.  You
...
+
+        ./tools/update_ipxe <ISSUE_ID>
+	git add tftpboot
+	git commit
+	git push -u origin HEAD:<branchname>
+
+    And again, Jenkins will automatically create a build and push it to
+    the experimental channel.
+
+You're done. After testing, get code review, integration approval, then merge.
```
Comment (jlevon) [inline README:85]: We should mention that you'll need help from an admin to allow a push of the multiple changesets

**Review by jlevon — CHANGES_REQUESTED** [review]
This looks great, just minor comments on the README

File: README (hunk 2 above)
Comment (bahamat) [reply README:98]: For vmware, our sample compute nodes use the virtualized hardware pxe. Do we normally test with vmware using two triton headnode images?

File: README (hunk 3 above)
Comment (bahamat) [reply README:85]: Something like this here?

> Note: You need to have write permission to Joyent's repo(s) for this step. If you're an outside contributor push it to your own fork and open a pull request.

Comment (jlevon) [reply README:85]: No, even normal ones like us will need get the settings changed as the repo should be configured to only allow single-commit pushes

File: README (hunk 2 above)
Comment (jlevon) [reply README:98]: The h/w pxe should still be bootstrapping iPXE from sdc-booter shouldn't it?

File: README (hunk 3 above)
Comment (bahamat) [reply README:85]: Ok, I've discussed with @trentm and we'll enable both merge and rebase operations, and it'll be up to the person conducting the merge to make sure to do the right thing. I've added a comment to the readme about this.

In the future, we can work out a safer way to handle this, but this should suffice for now.

File: README (hunk 2 above)
Comment (bahamat) [reply README:98]: It pxe boots from the virtual hardware first, then chain loads ipxe from the network.

I've added that to the list of tests.

File: README (hunk 1 above)
Comment (bahamat) [reply README]: Fixed.

File: README [hunk 4]
```diff
@@ -26,3 +26,91 @@ systems.  This utility is used by util/genliso and util/gensdsk.  You
...
+    the experimental channel.
+
+    Note: You need to have write permission to Joyent's repo(s) for this step.
+    If you're an outside contributor push it to your own fork and open a pull
+    request.
+
+You're done. After testing, get code review, integration approval, then merge.
+
+**Note about merging:** If you are simply syncing from upstream, be sure to
+rebase and merge. If you are adding a Joyent specific changet hat doesn't
```
Comment (jlevon) [inline README]: s/t hat/ that/

**Review by jlevon — CHANGES_REQUESTED** [review]
Just one more typo. Also, please put up a PR for sdc-booter and we'll get merged too. Thanks.

File: README (hunk 4 above)
Comment (bahamat) [reply README]: Thanks, fixed.

**Review by jlevon — APPROVED** [review]

**Comment by jlevon** [comment]
When you have a matching sdc-booter PR I'll add IA for both. Thanks!

**Comment by bahamat** [comment]
joyent/sdc-booter#12

**Comment by jlevon** [comment]
Oops, no IA here. Seems like you're good to go!

**Pull request merged**